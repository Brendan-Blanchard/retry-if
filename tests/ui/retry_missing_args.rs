use retrys::retry;
use retrys::ExponentialBackoffConfig;
use std::time::Duration;

const BACKOFF_CONFIG: ExponentialBackoffConfig = ExponentialBackoffConfig {
    max_retries: 5,
    t_wait: Duration::from_secs(1),
    backoff: 2.0,
    t_wait_max: None,
    backoff_max: None,
};

fn should_retry(_: ()) -> bool {
    true
}

#[retry]
fn some_method() -> () {}

fn main() {}
