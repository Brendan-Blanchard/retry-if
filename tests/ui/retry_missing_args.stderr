error: custom attribute panicked
  --> tests/ui/retry_missing_args.rs:16:1
   |
16 | #[retry]
   | ^^^^^^^^
   |
   = help: message: retry macro requires arguments of ExponentialBackoffConfig and 'should retry' function: Error("unexpected end of input, expected identifier")
